src/backend/core/models.py

```diff
@@ -463,6 +474,42 @@ def is_leaf(self):
         """
         return not self.has_deleted_children and self.numchild == 0
 
+    def get_ancestors(self):
+        """
+        :returns: A queryset containing the current node object's ancestors,
+            starting by the root node and descending to the parent.
+        """
+        if self.is_root():
+            return Document.objects.none()
+
+        paths = [self.path[0:pos] for pos in range(0, len(self.path), self.steplen)[1:]]
+        return (
+            Document.objects.select_related("creator")
+            .filter(path__in=paths)
+            .order_by("depth")
+        )
```

Comment:
Hmmm.... `range(0, len(self.path), self.steplen)[1:]` is actually `range(self.steplen, len(self.path), self.steplen)`?